### ISSUE TITLE
Region protection service API

### ISSUE THREAD

**Issue by NonSwag** [TheNextLvl-net/service-io#38]
### What feature do you want to see added?

A region protection service API that checks whether players or entities can do certain things inside specific areas

With Support for:
- [Protect](https://github.com/TheNextLvl-net/protect)
- [WorldGuard](https://github.com/EngineHub/WorldGuard)
- [CoreProtect](https://github.com/PlayPro/CoreProtect)
- [GriefPrevention](https://github.com/GriefPrevention/GriefPrevention)
- [LWC](https://github.com/pop4959/LWCX)
- [RedProtect](https://github.com/FabioZumbi12/RedProtect)

### Are there any alternatives?

-

### Anything else?

Basically [Protect's ProtectionService](https://github.com/TheNextLvl-net/protect/blob/master/api/src/main/java/net/thenextlvl/protect/service/ProtectionService.java)
